### PR TITLE
Make README more newbie friendly

### PR DESCRIPTION
**Pull request by osg**
Not sure how to test to see if this link behaves properly...

### CODE REVIEW

**Comment by osg** [comment]
Note that I omitted one book for the purpose of eliminating one extra scroll.

File: README.md [hunk 1]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+    }
+  ]
+  ```
 
-### In The Wild
+### Alternatives
 * [tobi/highlights](https://github.com/tobi/highlights) - Download your Kindle highlights and email random ones to your inbox
 
-### Contributing to kindle-highlights (PR's welcome)
+### Contributing to kindle-highlights (PRs welcome)
```
Comment (osg) [inline README.md:84]: Refer to http://www.grammarbook.com/punctuation/apostro.asp.

**Comment by osg** [comment]
Happy to make tweaks based on your feedback, @speric.

Already received feedback from a friend that this is not intended to be run within `irb`. However, it is indeed useful to see if it is working, so perhaps we can include both. Again, focusing on step-by-step for a newbie.

File: README.md [hunk 2]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
 
-### Use
-```ruby
-require 'kindle_highlights'
+### Usage
 
-# pass in your Amazon credentials. Loads your books (not highlights)
-# on init, so might take a while
-kindle = KindleHighlights::Client.new("email.address@gmail.com", "password")
+1. In an interactive Ruby shell ([`irb`](http://ruby-doc.org/stdlib-2.0.0/libdoc/irb/rdoc/IRB.html)), run the first of several commands:
```
Comment (denisdefreyne) [inline README.md]: Since this is a library, it’s not intended to be run with irb. (It _can_ be, but mostly for playing around.) I’d rephrase this as:

``````
1. Load the library:

    ```
    require 'kindle_highlights'
    ```
``````

File: README.md [hunk 3]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
-    "asin"          => "B005CQ2ZE6",
-    "customerId"    => "...",
-    "embeddedId"    => "From_the_Garden_to_the_City:420E805A",
-    "endLocation"   => 29591,
-    "highlight"     => "One of the most dangerous things you can believe in this world is that technology is neutral.",
-    "howLongAgo"    => "1 year ago",
-    "startLocation" => 29496,
-    "timestamp"     => 1320901233000
-  },
+1. Provide your Amazon credentials to access your book titles:
```
Comment (denisdefreyne) [inline README.md:19]: “Construct a `KindleHighlights::Client` with your Amazon credentials:”

File: README.md [hunk 4]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+
+   `kindle = KindleHighlights::Client.new("email.address@gmail.com", "password")`
+   
+   This might take a while.
```
Comment (denisdefreyne) [inline README.md]: Needs details on why this takes a while. “This fetches information for all Kindle books, and might therefore take a while.”

File: README.md [hunk 5]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+   
+1. Get each book (where the key is the [ASIN](http://www.amazon.com/gp/seller/asin-upc-isbn-info.html)), and its corresponding title:
```
Comment (denisdefreyne) [inline README.md:27]: Suggestion: “Call `#books` to get a hash of all books, where the key is the ASIN and the value the book title.”

Otherwise, it looks like a key/ASIN is somehow needed to get the list of books.

File: README.md [hunk 6]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+    "B009D6AGOM" => "The Rare Jewel of Christian Contentment",
+    "B00E25KVLW" => "Ruby on Rails 4.0 Guide",
+    "B004X5RLBY" => "The Seven Lamps of Architecture",
+    "B0032UWX1O" => "The Westminster Confession of Faith",
+    "B0026772N8" => "Zen and the Art of Motorcycle Maintenance"
   }
-]
-```
+    ```
+1. (Optional) If you are not sure how to produce human-readable output as in the preceding example, refer to [Best way to pretty print a hash]( http://stackoverflow.com/questions/8842546/best-way-to-pretty-print-a-hash).
```
Comment (denisdefreyne) [inline README.md]: I’d remove this.

File: README.md [hunk 7]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+1. Get your highlights for a specific book by using its ASIN:
```
Comment (denisdefreyne) [inline README.md:50]: “… by calling `#highlights_for`, passing in the book’s ASIN:”

File: README.md [hunk 8]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+      "highlight"     => "While God's words are eternal and unchanging, the tools we use to access those words do change, and those changes in technology also bring subtle changes to the practice of worship. When we fail to recognize the impact of such technological change, we run the risk of allowing our tools to dictate our methods. Technology should not dictate our values or our methods. Rather, we must use technology out of our convictions and values.",
+      "howLongAgo"    => "1 year ago",
+      "startLocation" => 53780,
+      "timestamp"     => 1321038422000
+    }
+  ]
+  ```
 
-### In The Wild
+### Alternatives
```
Comment (denisdefreyne) [inline README.md]: This list collects projects using `kindle-highlights`; it’s not a list of alternatives.

File: README.md [hunk 9]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+   `kindle = KindleHighlights::Client.new("email.address@gmail.com", "password")`
+   
+   This might take a while.
+   
+1. Get each book (where the key is the [ASIN](http://www.amazon.com/gp/seller/asin-upc-isbn-info.html)), and its corresponding title:
+
+   `kindle.books`
+   
+   Example output is as follows:
+    ```
```
Comment (denisdefreyne) [inline README.md]: Add the programming language after the three backticks, so that it gets highlighted properly:

``````
    ```ruby
    {
        "B002JCSCO8" => "The Art of the Commonplace: The Agrarian Essays of Wendell Berry",
``````

File: README.md [hunk 10]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
-```
+    ```
+1. (Optional) If you are not sure how to produce human-readable output as in the preceding example, refer to [Best way to pretty print a hash]( http://stackoverflow.com/questions/8842546/best-way-to-pretty-print-a-hash).
+1. Get your highlights for a specific book by using its ASIN:
+
+  `kindle.highlights_for("B005CQ2ZE6")`
+  
+  Example output is as follows:
+
+  ```
```
Comment (denisdefreyne) [inline README.md]: Add the programming language here too.

**Comment by osg** [comment]
Thanks, @ddfreyne! Will come back to this tomorrow after some sleep.

File: README.md (hunk 1 above)
Comment (speric) [inline README.md:84]: I fixed this typo on `master`; thanks!

Comment (osg) [inline README.md:84]: Not sure how to resolve the merge conflict that derives from this change...

File: README.md (hunk 3 above)
Comment (opensourcegrrrl) [inline README.md]: Less newbie friendly, but ok.

Comment (osg) [inline README.md:19]: Why does the user need to know that it creates a `KindleHighlights::Client`?

**Comment by osg** [comment]
Removed [WIP]. This is ready for final review.

**Comment by osg** [comment]
Please walk me through resolving this merge conflict.

**Comment by denisdefreyne** [comment]
I can help you out later.

File: README.md [hunk 11]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
-    "highlight"     => "One of the most dangerous things you can believe in this world is that technology is neutral.",
-    "howLongAgo"    => "1 year ago",
-    "startLocation" => 29496,
-    "timestamp"     => 1320901233000
-  },
+### Usage
+
+1. Load the library:
+
+   `require 'kindle_highlights'`
```
Comment (denisdefreyne) [inline README.md]: Prefer triple backticks for marking up code blocks:

``````
    ```
    require 'kindle_higlights'
    ```
``````

(same elsewhere)

File: README.md [hunk 12]
```diff
@@ -3,66 +3,79 @@ kindle-highlights
...
+
+1. Provide your Amazon credentials to access your book titles:
```
Comment (denisdefreyne) [inline README.md:19]: Provide your Amazon credentials how?

“to access your book titles” is not the point of creating a `KindleHighlights::Client`, so I’d leave that out.

File: README.md (hunk 3 above)
Comment (denisdefreyne) [inline README.md:19]: The user needs to explicitly create a `KindleHighlights::Client`, so the documentation should say so.

**Comment by osg** [comment]
Hi @speric, should I stop working on this PR and close it? Looks like the README is getting updated in parallel in `master` and I am not sure if the changes in my PR are for nought.

**Comment by speric** [comment]
@opensourcegrrrl I made a few changes in the gem itself that necessitated updating the README. This work had begun before you opened your PR. I'm sorry that there's now a merge conflict with your branch, but I couldn't release a new version of the gem without documenting the changes, as they were breaking changes.

Anyway, it's up to you if you want to rebase and work on the new README or not. I don't want to give you work to do, but neither do I want to discourage you from doing something you want to do!

I also would like to say that I appreciated your changes and the direction you were going. I did not realize until recently that many people use this gem who don't write Ruby, and I assumed too much familiarity with Ruby in the README.

I am going to make further changes to the README in the next day or two that are directly influenced by this PR. Specifically, I liked the way you moved explanations of what each method does, and what it returns, out of a comment within the Markdown code block. It reads better than the way I had it. Also, I now capture one of the mysterious errors you mentioned in another issue, and print out a readable explanation of what happened. In a future change to the README I will document how to work around that error.

So, thank you!

**Comment by osg** [comment]
@speric Ok, great. If you can wrap up your changes by Sat night, say, I can hopefully incorporate all of the feedback from Denis by Sunday night Berlin time. If not, that is okay too.

Kinda want to wrap things up as "good enough" before heading on vacation to China for about ten days. I anticipate not being on my ’puter as much, and my internet connections are likely to be spotty.

Also, I can easily just create a new branch after your changes, and thereby forego dealing with merge conflicts. :)

**Comment by osg** [comment]
Update: I have a healing fractured hand, and will need to revisit this in May. If you don't mind, please just keep this open so I can come back to it.

**Comment by speric** [comment]
@opensourcegrrrl I am going to close this for now; feel free to re-open it in the future.

**Comment by osg** [comment]
Yeah, no problem. Will need to come back to this due to priorities. Thanks for the note!
